command/command_test.go

```diff
@@ -0,0 +1,52 @@
+package command
+
+import (
+	"fmt"
+	"os"
+	"testing"
+)
+
+func TestExecuteProgram(t *testing.T) {
+
+	test := ExternalCommand{
+		Path: "./test_command",
+		Env:  os.Environ(),
+	}
+
+	programstate, err := ExecuteProgram(test, testErrorHandler)
+	if err != nil {
+		t.Errorf("Encountered error %s by ExecuteProgram()", err.Error())
+	}
+	out := programstate.Stdout()
+
+	serr := programstate.Stderr()
+	if len(serr) != 0 {
+		fmt.Println("Returned error by StderrNonBlocking()", string(serr))
+	}
+
+	want := "Hello"
+	got := string(out)
```

Comment:
See, it takes some time for your test program to actually output anything, so even here, you need to wait a while to check if the output is on the way.

Usually, the right way is to check the value of out() in a loop, till some timeout (you can have a helper that does it).

Reply:
Something like this:

```
timeout := time.After(1 * time.Second)

for want != got {
    select {
    case <-timeout:
        t.Errorf("Got text = %q, want %q", got, want)
    default:
    }
    got = string(programstate.Stdout())
}
```